.github/workflows/testing-and-deployment.yml

```
@@ -21,21 +21,22 @@ env:
 
 jobs:
   macOS:
-    # https://github.com/pyvista/pyvista/pull/5960
-    runs-on: macos-12
     name: Mac OS Unit Testing
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.9", "3.11"]
+        config:
+          - { os: macos-12, python-version: 3.9 } # intel processor
+          - { os: macos-latest, python-version: 3.12 } # arm processor
+    runs-on: ${{ matrix.config.os }}
```

Not sure if there's a reason we only have two configs in the matrix, should we add more? E.g. windows testing has a runner for every supported python version
```suggestion
        config:
          - { os: macos-12, python-version: 3.9 } # intel processor
          - { os: macos-13, python-version: 3.10 } # intel processor
          - { os: macos-14, python-version: 3.11 } # arm processor
          - { os: macos-latest, python-version: 3.12 } # arm processor
    runs-on: ${{ matrix.config.os }}
```